File: src/main/java/com/pubnub/api/models/consumer/pubsub/PNSignalResult.java
package com.pubnub.api.models.consumer.pubsub;

import com.google.gson.JsonElement;

import lombok.ToString;

@ToString(callSuper = true)
public class PNSignalResult extends MessageResult {

    public PNSignalResult(BasePubSubResult basePubSubResult, JsonElement message) {
        super(basePubSubResult, message);
    }
}

File: src/main/kotlin/com/pubnub/api/models/consumer/pubsub/PNSignalResult.kt
package com.pubnub.api.models.consumer.pubsub

import com.google.gson.JsonElement
import com.pubnub.api.SpaceId
import com.pubnub.api.callbacks.SubscribeCallback

/**
 * Wrapper around a signal received in [SubscribeCallback.signal].
 */
data class PNSignalResult internal constructor(
    private val basePubSubResult: PubSubResult,
    override val message: JsonElement,
    override val spaceId: SpaceId?,
    override val type: String?
) : MessageResult, PubSubResult by basePubSubResult
